libcasr/src/csharp.rs

```diff
@@ -133,34 +133,234 @@ impl ParseStacktrace for CSharpStacktrace {
 
         if let Some(s) = iter.peek() {
             let re = Self::get_regex_format_for_stacktrace_entry(s)?;
-
-            return iter.map(|s| Self::parse_stacktrace_entry(s, &re)).collect()
+            return iter.map(|s| Self::parse_stacktrace_entry(s, &re)).collect();
         }
 
         return std::iter::empty::<Result<StacktraceEntry>>().collect();
     }
 }
 
-/// Structure provides an interface for parsing c sharp exception message.
+/// Structure provides an interface for parsing c# exception message.
 pub struct CSharpException;
 
 impl Exception for CSharpException {
     fn parse_exception(stream: &str) -> Option<ExecutionClass> {
-        let re_mono = Regex::new(r"(?m)^Unhandled Exception:\n((?:.|\n)+?(?: ---> (?:.|\n)+?)*?)\n\s*(?:at .+|--- End of inner exception stack trace ---)$").unwrap();
-        let re_dotnet = Regex::new(r"(?m)^Unhandled exception\. ((?:.|\n)+?(?:\n ---> (?:.|\n)+?)*?)\n\s*(?:at .+|--- End of inner exception stack trace ---)$").unwrap();
+        let re = Regex::new(r"(?m)^Unhandled (e|E)xception(\. |:\n)((?:.|\n)*?)\s*(?:at [\S ]+|--- End of inner exception stack trace ---)$").unwrap();
+
+        let cap = re.captures(stream)?;
+        let get_group_as_str = |i| cap.get(i).unwrap().as_str();
 
-        let description = re_mono.captures(stream)
-            .or(re_dotnet.captures(stream))?
-            .get(1)?
-            .as_str();
+        let is_mono = get_group_as_str(1) == "E";
```

Comment:
As we discussed, we can have both variants of delimiters for both formats. In this case, we needn't now determine format in this function as it doesn't affect anything. I suggest make regex easier, remove groups for format recognizing and just parse every stream passed to the function using same delimiter " ---> "